File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/observe/ObserveLockPositionUseCase.java
package com.lattis.ellipse.domain.interactor.lock.observe;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/27/17.
 */

public class ObserveLockPositionUseCase extends BaseLockUseCase<Lock.Hardware.Position> {

    @Inject
    ObserveLockPositionUseCase(ThreadExecutor threadExecutor,
                               PostExecutionThread postExecutionThread,
                               BluetoothRepository bluetoothRepository,
                               LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread, bluetoothRepository,lockRepository);
    }

    public ObserveLockPositionUseCase forLock(Lock lock) {
        setScannedLock(lock);
        return this;
    }

    @Override
    protected Observable<Lock.Hardware.Position> buildUseCaseObservable() {
        return getLockPositionObservable();
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/observe/ObserveLockPositionUseCase.kt
package com.lattis.domain.usecase.lock.observe

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.models.Lock
import com.lattis.domain.repository.SaSOrPSLockRepository
import com.lattis.domain.usecase.lock.connect.ConnectToLockUseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject


class ObserveLockPositionUseCase @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository,
    lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<Lock.Hardware.Position>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository,
    lockRepository,
    saSOrPSLockRepository
) {

    fun withLockVendor(vendor:LockVendor): ObserveLockPositionUseCase {
        this.lockVendor = vendor
        return this
    }

    fun forLock(lock: Lock): ObserveLockPositionUseCase {
        setScannedLock(lock)
        return this
    }

    override fun buildUseCaseObservable(): Observable<Lock.Hardware.Position> {
        return lockPositionObservable(lockVendor)
    }
}